staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/validation/validation.go

```diff
@@ -194,8 +194,8 @@ func ConvertJSONSchemaPropsWithPostProcess(in *apiextensions.JSONSchemaProps, ou
 		}
 	}
 
-	if in.XPreserveUnknownFields {
-		out.VendorExtensible.AddExtension("x-kubernetes-preserve-unknown-fields", true)
+	if in.XPreserveUnknownFields != nil {
```

Comment:
`&& *in.XPreserveUnknownFields`?

Reply:
Validation enforces `true` if non-nil. Doesn't make much of a difference. In my variant even the non-validating `false` goes through.